Combined TPM eventlog

SergiiDmytruk opened the pull request:
Actual code changes are in EDK, coreboot only needs to not publish related ACPI entries.

~~Testing on APUs won't work without reverting modifications in [src/drivers/pc80/tpm/tis.c](https://github.com/Dasharo/coreboot/commit/9a5e8dbee3eb37b026bc797fbcb958035be1c89c#diff-aa309ea2137f2e8e574b38749cd1abee7ee6bc7ea195ad3a6522fd9bcbbb0392) which somehow disables a working TPM.~~

EDK PR: https://github.com/Dasharo/edk2/pull/139

This now also includes picking TCG log format at runtime.

SergiiDmytruk committed:
acpi: don't publish TPM logs if EDK payload is used

EDK will publish its own version of the log after parsing and
importing coreboot's log discovered through CBMEM.

Publishing is done by appending a table, so coreboot must avoid
adding corresponding log tables to let OS find a more complete one
from EDK.

Upstream-Status: Pending
Change-Id: Iec92c2b5c426ee003e81996937862d81cb4ead24
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
security/tpm: replace TPM_MEASURE_ALGO with tpm_log_alg()

No functional changes.  This replaces a macro with an inline function to
make code more readable and more convenient to extend in the future.

Upstream-Status: Pending
Change-Id: I456bc3bb749a9b58fba72f5562195525e55290bf
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

SergiiDmytruk committed:
security/tpm: add CONFIG_TPM_LOG_TCG

This event log format option automatically selects TCG log format
depending on which TPM is present.

Upstream-Status: Pending
Change-Id: I1997396f24ff6362fe64ac56f8e61efcf2ffb0f7
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

krystian-hebel reviewed (dismissed):
Approved, but edk2 must be merged first.

m-iwanicki commented:
@SergiiDmytruk Not sure if it's due to your PR but when I tested latest artifact: https://github.com/Dasharo/coreboot/actions/runs/9800367204/job/27062155366 on Protectli VP2420 `Dasharo System Features` menu disappeared.

krystian-hebel commented:
> @SergiiDmytruk Not sure if it's due to your PR but when I tested latest artifact: https://github.com/Dasharo/coreboot/actions/runs/9800367204/job/27062155366 on Protectli VP2420 `Dasharo System Features` menu disappeared.

Probably caused by lack of https://github.com/Dasharo/edk2/pull/148 after the bug was introduced in https://github.com/Dasharo/edk2/pull/144 / https://github.com/Dasharo/DasharoModulePkg/pull/50

SergiiDmytruk commented:
@m-iwanicki I saw that but assumed a messed up `.config`.  It's actually an issue fixed by https://github.com/Dasharo/edk2/commit/d130aececbdd5f50fa93d0e2b2fa52cee036788a commit which was missing on `rebased`.  Added it there and updated both PRs to include that fix.

miczyg1 commented:
> Testing on APUs won't work without reverting modifications in [src/drivers/pc80/tpm/tis.c](https://github.com/Dasharo/coreboot/commit/9a5e8dbee3eb37b026bc797fbcb958035be1c89c#diff-aa309ea2137f2e8e574b38749cd1abee7ee6bc7ea195ad3a6522fd9bcbbb0392) which somehow disables a working TPM.

This was supposed to fix it: https://github.com/Dasharo/coreboot/commit/d3b853142a213dfd5c7a21a0780ff94267b4ba38

Do we have some insights why it doesn't work/what is the problem exactly?

SergiiDmytruk commented:
> > Testing on APUs won't work without reverting modifications in [src/drivers/pc80/tpm/tis.c](https://github.com/Dasharo/coreboot/commit/9a5e8dbee3eb37b026bc797fbcb958035be1c89c#diff-aa309ea2137f2e8e574b38749cd1abee7ee6bc7ea195ad3a6522fd9bcbbb0392) which somehow disables a working TPM.
> 
> This was supposed to fix it: [d3b8531](https://github.com/Dasharo/coreboot/commit/d3b853142a213dfd5c7a21a0780ff94267b4ba38)
> 
> Do we have some insights why it doesn't work/what is the problem exactly?

This PR is older than that commit.

miczyg1 commented:
@macpijan do we have some working setups with TPM1.2 in the lab?

krystian-hebel commented:
IIRC one of KGPEs should still have TPM1.2 which was used for AEM. Note that it currently has ASUS BIOS, on 16Mb chip.

miczyg1 commented:
> IIRC one of KGPEs should still have TPM1.2 which was used for AEM. Note that it currently has ASUS BIOS, on 16Mb chip.

@krystian-hebel something that will run on this coreboot version :)

SergiiDmytruk commented:
Rebased: updated EDK2 hash which also fixed merge conflict.

miczyg1 commented:
Rebased on top of recent dasharo branch

SergiiDmytruk committed:
configs: bump Dasharo EDK2 to get combined TPM log

Upstream-Status: Inappropriate (Dasharo downstream)
Change-Id: Iacee23253d2766d9f3835860d0d64d84b5bcd880
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>

miczyg1 commented:
Can not FF-merge it, not up to date with dasharo branch
